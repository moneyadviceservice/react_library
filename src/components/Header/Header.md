### Renders an identical copy of the RAD header.

```jsx
import React, { useState } from 'react';
import { Header, Row, Button } from '@moneypensionservice/directories';

const [lng, changeLng] = useState('en');

<>
  <Row margin={{bottom: '20px'}}>
    <Button onClick={() => changeLng('en')}>English</Button>
    <Button onClick={() => changeLng('cy')}>Welsh</Button>
  </Row>
  <Header 
    currentLng={lng} 
    setLng={() => changeLng(lng === 'en' ? 'cy' : 'en')} />
</>
```
