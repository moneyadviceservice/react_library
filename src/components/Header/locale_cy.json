{
  "mapsLogo": {
    "key1": "Gwasanaeth Cynghori Ariannol",
    "key2": "yn cael ei ddarparu gan"
  },
  "localeButton": "English"
}
