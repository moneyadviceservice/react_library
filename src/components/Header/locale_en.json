{
  "mapsLogo": {
    "key1": "The Money Advice",
    "key2": "Service is provided by"
  },
  "localeButton": "Cymraeg"
}
