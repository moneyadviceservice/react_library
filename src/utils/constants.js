export const breakpoints = {
  xs: 1,
  sm: 48, // 768px
  md: 64, // 1024px
  lg: 90, // 1440px
  xl: 120, // 1920px
}

export const dimensions = ['xs', 'sm', 'md', 'lg', 'xl']

export const mediaQuery = 'only screen'

export const ALIGN_SELF_MAP = {
  center: 'center',
  end: 'flex-end',
  start: 'flex-start',
  stretch: 'stretch',
}

export const SPACING_SIZES = [
  'xxsmall',
  'xsmall',
  'small',
  'medium',
  'large',
  'xlarge',
]

