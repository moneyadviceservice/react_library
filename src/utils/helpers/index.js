export * from './background'
export * from './responsive'
export * from './flexbox'
export * from './styles'
export * from './mediaQueries'
export * from './sizing'
export * from './refs'
