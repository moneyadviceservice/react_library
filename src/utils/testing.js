import { render, act } from '@testing-library/react'
import GlobalTheme from '../components/ThemeProvider'

const customRender = (ui, options) =>
  render(ui, { wrapper: GlobalTheme, ...options })

const waitRender = async (ui, options) => {
  const { container, ...rest } = customRender(ui, options)

  await act(async () => {
    await Promise.resolve(container)
  })

  return { container, ...rest }
}

export * from '@testing-library/react'

export { customRender as render, waitRender }
