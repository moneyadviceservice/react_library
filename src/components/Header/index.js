import React from 'react'
import PropTypes from 'prop-types'
import { genericPropTypes, genericPropsDefaults } from '../../utils/prop-types'
import {
  HeaderContainer,
  HeaderRow,
  LogoContainer,
  LocaleContainer,
  MapsBannerRow,
  MapsLogo,
  MapsLogoText,
} from './StyledHeader'
import { Col } from '../Grid'
// logos
import MasLogoEng from '../../assets/Images/mas_logo_en.svg'
import MasLogoCy from '../../assets/Images/mas_logo_cy.svg'
import MapsLogoEng from '../../assets/Images/maps_logo_en.svg'
import MapsLogoCy from '../../assets/Images/maps_logo_cy.svg'
// translations
import LocaleEn from './locale_en.json'
import LocaleCy from './locale_cy.json'

const Header = ({
  a11yTitle,
  as,
  children,
  currentLng,
  lngUrl,
  i18nLng,
  setLng,
  ...rest
}) => {
  // locale
  const lng = i18nLng || (currentLng === 'cy' ? LocaleCy : LocaleEn)

  return (
    <>
      <HeaderContainer fluid forwardedAs={as} {...rest}>
        <MapsBannerRow justify="center">
          <Col direction="row" justify="flex-end" align="center" constrained>
            <MapsLogo
              grow={false}
              direction="row"
              justify="flex-end"
              align="center"
            >
              <MapsLogoText>
                <span>{lng.mapsLogo.key1}</span>
                <span>{lng.mapsLogo.key2}</span>
              </MapsLogoText>
              {currentLng === 'en' ? <MapsLogoEng /> : <MapsLogoCy />}
            </MapsLogo>
          </Col>
        </MapsBannerRow>
        <HeaderRow justify="center" align="center">
          <Col
            direction="row"
            justify="space-between"
            align="center"
            constrained
          >
            <Col grow={false}>
              <LogoContainer
                a11yTitle="Money Advice Service"
                href="https://www.moneyadviceservice.org.uk/en"
                margin="0"
              >
                {currentLng === 'cy' ? <MasLogoCy /> : <MasLogoEng />}
              </LogoContainer>
            </Col>
            <Col grow={false}>
              <LocaleContainer
                color="white"
                textSize="0.875rem"
                weight="500"
                margin="0"
                onClick={setLng}
                href={lngUrl}
              >
                {lng.localeButton}
              </LocaleContainer>
            </Col>
          </Col>
        </HeaderRow>
      </HeaderContainer>
      {children}
    </>
  )
}

// Documentation
Header.propTypes = {
  /** Current Language Value */
  currentLng: PropTypes.oneOf(['en', 'cy']),
  /** Server-side fallback url to change language */
  lngUrl: PropTypes.string,
  /** Alternate translations for header. */
  i18nLng: PropTypes.object,
  /** Function to trigger by Locale Button */
  setLng: PropTypes.func,
  ...genericPropTypes,
}

Header.defaultProps = {
  as: 'header',
  currentLng: 'en',
  ...genericPropsDefaults(),
}

/** @component */
export { Header }
