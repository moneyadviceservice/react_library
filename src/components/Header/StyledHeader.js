import styled from 'styled-components'
import { resolveMedia } from '../../utils/helpers'
import Container from '../Grid/Container'
import { Anchor } from '../Anchor/index'

const HeaderContainer = styled(Container)`
  background: ${props =>
    !props.background && props.theme.colors.header.background};
  min-height: 3.375rem;

  ${resolveMedia.md`
    min-height: 4.125rem;
  `}
`

const LocaleContainer = styled(Anchor)``

const LogoContainer = styled(Anchor)`
  overflow: hidden;
  width: 200px;
  & svg {
    fill: white;
  }
  ${resolveMedia.md`
    width: 285px;
  `}
`

export { HeaderContainer, LogoContainer, LocaleContainer }
