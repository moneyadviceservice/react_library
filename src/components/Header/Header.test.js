import React from 'react'
import { render, fireEvent } from '../../utils/testing'
import { Header } from '../Header'

describe('<Header />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Header />)
      expect(container).toMatchSnapshot()
    })
    it('should match snapshot in Welsh', () => {
      const { container } = render(<Header currentLng="cy" />)
      expect(container).toMatchSnapshot()
    })
  })

  describe('lngUrl prop', () => {
    it('should change the anchor url', () => {
      const { container, getByTestId } = render(
        <Header lngUrl="https://www.test.com/" />
      )
      // snapshot
      expect(container).toMatchSnapshot()
      // href
      expect(getByTestId('localeAnchor').href).toBe('https://www.test.com/')
    })
  })

  describe('setLng prop', () => {
    it('should call function on anchor click', () => {
      const onClickMock = jest.fn()
      const { container, getByTestId } = render(<Header setLng={onClickMock} />)
      // snapshot
      expect(container).toMatchSnapshot()

      fireEvent.click(getByTestId('localeAnchor'))

      expect(onClickMock).toHaveBeenCalledTimes(1)
    })
  })
})
