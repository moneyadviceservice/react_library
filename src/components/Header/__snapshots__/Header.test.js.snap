// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Snapshots should match snapshot 1`] = `
<div>
  <header
    class="styledContainer__ContainerWrapper-ux129m-0 cBvoAu StyledHeader__HeaderContainer-nhaq8b-0 kqbXpf"
  >
    <div
      class="styledRow__RowWrapper-sc-1ip85in-0 XbhRj StyledHeader__MapsBannerRow-nhaq8b-4 gZEGHo"
    >
      <div
        class="styledCol__ColWrapper-sc-1mivnjm-0 echTSp"
      >
        <div
          class="styledCol__ColWrapper-sc-1mivnjm-0 hwdKXn StyledHeader__MapsLogo-nhaq8b-5 jhPcbu"
        >
          <span
            class="StyledHeader__MapsLogoText-nhaq8b-6 gqpbPZ"
          >
            <span>
              The Money Advice
            </span>
            <span>
              Service is provided by
            </span>
          </span>
          <test-file-stub />
        </div>
      </div>
    </div>
    <div
      class="styledRow__RowWrapper-sc-1ip85in-0 cYiCXW StyledHeader__HeaderRow-nhaq8b-1 bGFUQm"
    >
      <div
        class="styledCol__ColWrapper-sc-1mivnjm-0 hYHTJn"
      >
        <div
          class="styledCol__ColWrapper-sc-1mivnjm-0 kcBKMp"
        >
          <a
            aria-label="Money Advice Service"
            class="StyledParagraph-sc-1toaej-0 cBLpIS StyledAnchor-sc-1qj1ci5-0 fNhjgf StyledHeader__LogoContainer-nhaq8b-3 jcJypd"
            href="https://www.moneyadviceservice.org.uk/en"
          >
            <test-file-stub />
          </a>
        </div>
        <div
          class="styledCol__ColWrapper-sc-1mivnjm-0 kcBKMp"
        >
          <a
            class="StyledParagraph-sc-1toaej-0 hZKqQo StyledAnchor-sc-1qj1ci5-0 jjWyuo StyledHeader__LocaleContainer-nhaq8b-2"
            data-testid="localeAnchor"
          >
            Cymraeg
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`<Header /> Snapshots should match snapshot in Welsh 1`] = `
<div>
  <header
    class="styledContainer__ContainerWrapper-ux129m-0 cBvoAu StyledHeader__HeaderContainer-nhaq8b-0 kqbXpf"
  >
    <div
      class="styledRow__RowWrapper-sc-1ip85in-0 XbhRj StyledHeader__MapsBannerRow-nhaq8b-4 gZEGHo"
    >
      <div
        class="styledCol__ColWrapper-sc-1mivnjm-0 echTSp"
      >
        <div
          class="styledCol__ColWrapper-sc-1mivnjm-0 hwdKXn StyledHeader__MapsLogo-nhaq8b-5 jhPcbu"
        >
          <span
            class="StyledHeader__MapsLogoText-nhaq8b-6 gqpbPZ"
          >
            <span>
              Gwasanaeth Cynghori Ariannol
            </span>
            <span>
              yn cael ei ddarparu gan
            </span>
          </span>
          <test-file-stub />
        </div>
      </div>
    </div>
    <div
      class="styledRow__RowWrapper-sc-1ip85in-0 cYiCXW StyledHeader__HeaderRow-nhaq8b-1 bGFUQm"
    >
      <div
        class="styledCol__ColWrapper-sc-1mivnjm-0 hYHTJn"
      >
        <div
          class="styledCol__ColWrapper-sc-1mivnjm-0 kcBKMp"
        >
          <a
            aria-label="Money Advice Service"
            class="StyledParagraph-sc-1toaej-0 cBLpIS StyledAnchor-sc-1qj1ci5-0 fNhjgf StyledHeader__LogoContainer-nhaq8b-3 jcJypd"
            href="https://www.moneyadviceservice.org.uk/en"
          >
            <test-file-stub />
          </a>
        </div>
        <div
          class="styledCol__ColWrapper-sc-1mivnjm-0 kcBKMp"
        >
          <a
            class="StyledParagraph-sc-1toaej-0 hZKqQo StyledAnchor-sc-1qj1ci5-0 jjWyuo StyledHeader__LocaleContainer-nhaq8b-2"
            data-testid="localeAnchor"
          >
            English
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`<Header /> lngUrl prop should change the anchor url 1`] = `
<div>
  <header
    class="styledContainer__ContainerWrapper-ux129m-0 cBvoAu StyledHeader__HeaderContainer-nhaq8b-0 kqbXpf"
  >
    <div
      class="styledRow__RowWrapper-sc-1ip85in-0 XbhRj StyledHeader__MapsBannerRow-nhaq8b-4 gZEGHo"
    >
      <div
        class="styledCol__ColWrapper-sc-1mivnjm-0 echTSp"
      >
        <div
          class="styledCol__ColWrapper-sc-1mivnjm-0 hwdKXn StyledHeader__MapsLogo-nhaq8b-5 jhPcbu"
        >
          <span
            class="StyledHeader__MapsLogoText-nhaq8b-6 gqpbPZ"
          >
            <span>
              The Money Advice
            </span>
            <span>
              Service is provided by
            </span>
          </span>
          <test-file-stub />
        </div>
      </div>
    </div>
    <div
      class="styledRow__RowWrapper-sc-1ip85in-0 cYiCXW StyledHeader__HeaderRow-nhaq8b-1 bGFUQm"
    >
      <div
        class="styledCol__ColWrapper-sc-1mivnjm-0 hYHTJn"
      >
        <div
          class="styledCol__ColWrapper-sc-1mivnjm-0 kcBKMp"
        >
          <a
            aria-label="Money Advice Service"
            class="StyledParagraph-sc-1toaej-0 cBLpIS StyledAnchor-sc-1qj1ci5-0 fNhjgf StyledHeader__LogoContainer-nhaq8b-3 jcJypd"
            href="https://www.moneyadviceservice.org.uk/en"
          >
            <test-file-stub />
          </a>
        </div>
        <div
          class="styledCol__ColWrapper-sc-1mivnjm-0 kcBKMp"
        >
          <a
            class="StyledParagraph-sc-1toaej-0 hZKqQo StyledAnchor-sc-1qj1ci5-0 jjWyuo StyledHeader__LocaleContainer-nhaq8b-2"
            data-testid="localeAnchor"
            href="https://www.test.com/"
          >
            Cymraeg
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`<Header /> setLng prop should call function on anchor click 1`] = `
<div>
  <header
    class="styledContainer__ContainerWrapper-ux129m-0 cBvoAu StyledHeader__HeaderContainer-nhaq8b-0 kqbXpf"
  >
    <div
      class="styledRow__RowWrapper-sc-1ip85in-0 XbhRj StyledHeader__MapsBannerRow-nhaq8b-4 gZEGHo"
    >
      <div
        class="styledCol__ColWrapper-sc-1mivnjm-0 echTSp"
      >
        <div
          class="styledCol__ColWrapper-sc-1mivnjm-0 hwdKXn StyledHeader__MapsLogo-nhaq8b-5 jhPcbu"
        >
          <span
            class="StyledHeader__MapsLogoText-nhaq8b-6 gqpbPZ"
          >
            <span>
              The Money Advice
            </span>
            <span>
              Service is provided by
            </span>
          </span>
          <test-file-stub />
        </div>
      </div>
    </div>
    <div
      class="styledRow__RowWrapper-sc-1ip85in-0 cYiCXW StyledHeader__HeaderRow-nhaq8b-1 bGFUQm"
    >
      <div
        class="styledCol__ColWrapper-sc-1mivnjm-0 hYHTJn"
      >
        <div
          class="styledCol__ColWrapper-sc-1mivnjm-0 kcBKMp"
        >
          <a
            aria-label="Money Advice Service"
            class="StyledParagraph-sc-1toaej-0 cBLpIS StyledAnchor-sc-1qj1ci5-0 fNhjgf StyledHeader__LogoContainer-nhaq8b-3 jcJypd"
            href="https://www.moneyadviceservice.org.uk/en"
          >
            <test-file-stub />
          </a>
        </div>
        <div
          class="styledCol__ColWrapper-sc-1mivnjm-0 kcBKMp"
        >
          <a
            class="StyledParagraph-sc-1toaej-0 hZKqQo StyledAnchor-sc-1qj1ci5-0 jjWyuo StyledHeader__LocaleContainer-nhaq8b-2"
            data-testid="localeAnchor"
          >
            Cymraeg
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;
